code/game/verbs/suicide.dm

```diff
@@ -1,302 +1,281 @@
+/mob/living/proc/attempt_item_suicide(var/obj/item/suicide_item)
 
-					if(3, 5, 7, 9, 10, 12) //2 damage types
-						damage_mod = 2
+	if(suicide_item) //We need the item to be there to begin, otherwise abort
+		var/damagetype = suicide_item.suicide_act(src)
+		if(damagetype)
+			var/damage_mod = 1
+			switch(damagetype) //Sorry about the magic numbers.
+							   //brute = 1, burn = 2, tox = 4, oxy = 8
+				if(15) //4 damage types
+					damage_mod = 4
 
-					if(1, 2, 4, 8) //1 damage type
-						damage_mod = 1
+				if(6, 11, 13, 14) //3 damage types
+					damage_mod = 3
 
-					else //This should not happen, but if it does, everything should still work
-						damage_mod = 1
+				if(3, 5, 7, 9, 10, 12) //2 damage types
+					damage_mod = 2
 
-				//Do 175 damage divided by the number of damage types applied.
-				if(damagetype & BRUTELOSS)
-					adjustBruteLoss(175/damage_mod)
+				if(1, 2, 4, 8) //1 damage type
+					damage_mod = 1
 
-				if(damagetype & FIRELOSS)
-					adjustFireLoss(175/damage_mod)
+				else //No special suicide exists for this item or something fucked up, abort and go for a normal suicide instead
+					return
 
-				if(damagetype & TOXLOSS)
-					adjustToxLoss(175/damage_mod)
+			//Do 175 damage divided by the number of damage types applied.
+			if(damagetype & BRUTELOSS)
+				adjustBruteLoss(175/damage_mod)
 
-				if(damagetype & OXYLOSS)
-					adjustOxyLoss(175/damage_mod)
+			if(damagetype & FIRELOSS)
+				adjustFireLoss(175/damage_mod)
 
-				//If something went wrong, just do normal oxyloss
-				if(!(damagetype | BRUTELOSS) && !(damagetype | FIRELOSS) && !(damagetype | TOXLOSS) && !(damagetype | OXYLOSS))
-					adjustOxyLoss(max(175 - getToxLoss() - getFireLoss() - getBruteLoss() - getOxyLoss(), 0))
+			if(damagetype & TOXLOSS)
+				adjustToxLoss(175/damage_mod)
 
-				updatehealth()
-				return
+			if(damagetype & OXYLOSS)
+				adjustOxyLoss(175/damage_mod)
 
+			updatehealth()
+			return 1
+
+/mob/living/carbon/human/attempt_suicide(forced = 0, suicide_set = 1)
+
+	if(!forced)
+
+		var/confirm = alert("Are you sure you want to commit suicide? This action cannot be undone and you will not able to be revived.", "Confirm Suicide", "Yes", "No")
+
+		if(confirm != "Yes")
+			return
 
+		if(stat != CONSCIOUS)
+			to_chat(src, "<span class='warning'>You can't commit suicide in this state!</span>")
+			return
+
+		if(istype(wear_mask, /obj/item/clothing/mask/happy))
+			to_chat(src, "<span class='sinister'>BUT WHY? I'M SO HAPPY!</span>")
+			return
+
+		var/mob/living/simple_animal/borer/B = has_brain_worms()
+		if(B && B.controlling) //Borer
+			to_chat(src, "<span class='warning'>You cannot commit suicide, your host is clinging to life enough to resist it.</span>")
+			return
+
+		var/permitted = 1
+		var/list/allowed = list("Syndicate", "traitor", "Wizard", "Head Revolutionary", "Cultist", "Changeling")
+		for(var/T in allowed)
+			if(mind.special_role == T)
+				permitted = 1
+				break
+
+		if(!permitted)
```

Comment:
This code block also appears to have no effect since permitted is always true. I'm not quite sure what's going on here, but apparently it's not doing any harm, so